pyscriptjs/src/plugins/automaticImport.ts

```diff
@@ -0,0 +1,107 @@
+import type { AppConfig } from '../pyconfig';
+import { robustFetch } from '../fetch';
+import { Plugin } from '../plugin';
+import { getLogger } from '../logger';
+
+const logger = getLogger('automatic-import-plugin');
+
+export class AutomaticImportPlugin extends Plugin {
+    pyscriptTag = document.querySelector('py-script');
+
+    /**
+     * Trim whitespace from the beginning and end of a string.
+     * The text that we are getting has new lines and possibly
+     * whitespace at the beginning and end of the line. Since
+     * we want to match only import statements, we need to trim
+     * the whitespace on each line, join them together and then
+     * match each beginning of the line so the regex rule will
+     * only match what we perceive as import statements.
+     *
+     * NOTE: This is not a perfect solution and I'm a bit worried
+     * about the performance of this. I'm sure there is a better
+     * way to do this.
+     *
+     * @param text - The text to trim
+     */
+    _trimWhiteSpace(text: string): string {
+        const lines = text.split('\n');
+        return lines.map(line => line.trim()).join('\n');
+    }
+    /**
+     *
+     * Use regex magic to capture import statements and add the
+     * dependency(s) to the packages list in the config.
+     *
+     * @param text - The text to search for import statements
+     * @param config - The config object to add the dependencies to
+     *
+     */
+    _addImportToPackagesList(text: string, config: AppConfig) {
+        // Regex encantation to capture the imported dependency into a
+        // named group called "dependency". The rule will match any text
+        // that contains 'import' or 'from' at the beginning of the line
+        // or '\nimport', '\nfrom' which is the case when we invoke _trimWhiteSpace
+        const importRegexRule = /^(?:\\n)?(?:import|from)\s+(?<dependency>[a-zA-Z0-9_]+)?/gm;
```

Comment:
Can't you add `\s*` to the beginning of the regex and remove `trimWhiteSpace`?
```suggestion
        const importRegexRule = /^(?:\\n)?\s*(?:import|from)\s+(?<dependency>[a-zA-Z0-9_]+)?/gm;
```